src/main/java/org/zalando/nakadi/repository/zookeeper/ZooKeeperHolder.java

```diff
@@ -9,78 +8,68 @@
-        this.sessionTimeoutMs = sessionTimeoutMs;
-        this.connectionTimeoutMs = connectionTimeoutMs;
-
-        initExhibitor();
-    }
+        final EnsembleProvider ensembleProvider = createEnsembleProvider(zookeeperConnection);
 
-    private void initExhibitor() throws Exception {
-        final RetryPolicy retryPolicy = new ExponentialBackoffRetry(EXHIBITOR_RETRY_TIME, EXHIBITOR_RETRY_MAX);
-        final EnsembleProvider ensembleProvider;
-        if (exhibitorAddresses != null) {
-            final Collection<String> exhibitorHosts = Arrays.asList(exhibitorAddresses.split("\\s*,\\s*"));
-            final Exhibitors exhibitors = new Exhibitors(exhibitorHosts, exhibitorPort,
-                    () -> zookeeperBrokers + zookeeperKafkaNamespace);
-            final ExhibitorRestClient exhibitorRestClient = new DefaultExhibitorRestClient();
-            ensembleProvider = new ExhibitorEnsembleProvider(exhibitors,
-                    exhibitorRestClient, "/exhibitor/v1/cluster/list", EXHIBITOR_POLLING_MS, retryPolicy);
-            ((ExhibitorEnsembleProvider) ensembleProvider).pollForInitialEnsemble();
-        } else {
-            ensembleProvider = new FixedEnsembleProvider(zookeeperBrokers + zookeeperKafkaNamespace);
-        }
         zooKeeper = CuratorFrameworkFactory.builder()
                 .ensembleProvider(ensembleProvider)
-                .retryPolicy(retryPolicy)
+                .retryPolicy(new ExponentialBackoffRetry(EXHIBITOR_RETRY_TIME, EXHIBITOR_RETRY_MAX))
                 .sessionTimeoutMs(sessionTimeoutMs)
                 .connectionTimeoutMs(connectionTimeoutMs)
                 .build();
         zooKeeper.start();
     }
 
-    public CuratorFramework get() {
-        return zooKeeper;
-    }
-
-    private class ExhibitorEnsembleProvider extends org.apache.curator.ensemble.exhibitor.ExhibitorEnsembleProvider {
-
-        ExhibitorEnsembleProvider(final Exhibitors exhibitors, final ExhibitorRestClient restClient,
-                                  final String restUriPath, final int pollingMs, final RetryPolicy retryPolicy) {
-            super(exhibitors, restClient, restUriPath, pollingMs, retryPolicy);
+    private EnsembleProvider createEnsembleProvider(final ZookeeperConnection conn) throws Exception {
+        switch (conn.getType()) {
+            case EXHIBITOR:
+                final Exhibitors exhibitors = new Exhibitors(
+                        conn.getAddresses().stream().map(AddressPort::getAddress).collect(Collectors.toList()),
+                        conn.getAddresses().get(0).getPort(),
+                        () -> {
+                            throw new RuntimeException("There is no backup connection string (or it is wrong)");
+                        });
+                final ExhibitorRestClient exhibitorRestClient = new DefaultExhibitorRestClient();
+                final org.apache.curator.ensemble.exhibitor.ExhibitorEnsembleProvider result =
```

Comment:
maybe you can import this class

Reply:
True, I removed inner class with the same name, but forgot import